integration_test/mysql/test_helper.exs

```diff
@@ -43,7 +43,8 @@ defmodule Ecto.Integration.Case do
   end
 
   setup do
-    Ecto.Adapters.SQL.restart_test_transaction(TestRepo, [])
+    Ecto.Adapters.SQL.rollback_test_transaction(TestRepo, [])
+    Ecto.Adapters.SQL.begin_test_transaction(TestRepo, [])
```

Comment:
The modify and remove column tests that are being added with this PR require `CREATE` and `ALTER TABLE` statements to be issued to the database.  These statements will cause an implicit commit of any transactions or savepoints in MySQL.  This causes the test transaction that was being restarted here to be released, and any tests that are performed after the new tests will fail with the error "SAVEPOINT ecto_sandbox does not exist".  The above change provides virtually the same functionality as `restart_test_transaction` but does not result in MySQL errors.

I tried a few other changes to work around the error, but this seemed like the most straight-forward.  Is there a better work-around that I should implement instead?